addons/mail/static/src/core/web/mail_composer_attachment_list.js

```diff
@@ -20,7 +20,7 @@ export class MailComposerAttachmentList extends Many2ManyBinaryField {
     async onFileRemove(fileId) {
         super.onFileRemove(fileId);
         const attachment = this.mailStore.Attachment.insert(fileId);
-        if (attachment) {
+        if (attachment && !this.props.record.data.template_id[1].includes(fileId)) {
```

Comment:
Oh actually, you do need to check if there's a template otherwise you'll get an error whenever you unlink attachments from non-template messages :x